src/document_generation.py

```diff
@@ -0,0 +1,45 @@
+import streamlit as st
```

Comment:
Streamlit taucht in der pyproject.toml bisher nicht auf

Reply:
Zwecks Flexibilität wäre es aber zu überlegen, ob man frontend und backend nicht generell trennt.

[Bsp](https://medium.com/codex/streamlit-fastapi-%EF%B8%8F-the-ingredients-you-need-for-your-next-data-science-recipe-ffbeb5f76a92)

Reply:
Ich denke mal die API wird nicht allzu komplex und die REST Api wird auch nicht das Performance-Bottleneck sein. Anstatt [FastAPI](https://fastapi.tiangolo.com/) würde für sich deshalb auch das minimalistischere [flask](https://flask.palletsprojects.com/en/3.0.x/) anbieten.

Reply:
So könnten die Endpunkte beispielsweise aussehen:

```
POST /api/v1/generate { "limit": int } -> { "status": int, "response": str, "error_msg": str }
POST /api/v1/answer { "query": str, ["fact_checking": bool] } -> { "status": int, "response": str, "error_msg": str }
GET /api/v1/prompt -> { "status": int, "response": str, "error_msg": str }
POST /api/v1/prompt -> { "status": int, "error_msg": str }
```

Reply:
streamlit zu pyproject.toml hinzugefügt.
Bezüglich API wäre es nicht schlecht am Montag einmal darüber zu reden. Ich kann mir da leider noch kein Bild drüber machen, da ich selber noch keine Erfahrung damit habe.